rules/follow-semver-when-publishing-npm-packages/rule.md

```diff
@@ -0,0 +1,88 @@
+---
+type: rule
+title: Do you follow semver when publishing npm packages?
+seoDescription: When publishing npm packages, following Semantic Versioning (SemVer) is important for communicate changes clearly and maintain compatibility for your users. Learn how to manage versions responsibly and build trust with your user.  
+uri: follow-semver-when-publishing-npm-packages
+authors:
+  - title: Ben Neoh
+    url: https://www.ssw.com.au/people/ben-neoh/
+related:
+  - different-symbols-mean-for-npm-version
+  - packages-up-to-date
+created: 2024-12-5T08:33:21.000Z
+guid: 03d090d3-d813-43d9-998e-209c895050b3
+---
+
+When publishing an npm package, following **Semantic Versioning (SemVer)** is essential. It communicates changes clearly to your users and ensures smooth updates for their projects. This guide will tell you why adhering to SemVer is critical and how to do it effectively.  
+
+<!--endintro-->
+
+## Why use SEMVER in NPM Publishing
+
+Semantic Versioning (SemVer) helps your users understand the impact of updates and manage their own dependencies more effectively. By adhering to SemVer, you make it clear whether an update introduces breaking changes, new features, or just bug fixes:
+
+- **MAJOR (Breaking Changes):** Signals to users that there are incompatible changes.
+- **MINOR (New Features):** Informs users about new features that won’t break existing functionality.
+- **PATCH (Bug Fixes):** Indicates that bug fixes or small improvements have been made without changing behavior.
+
+Without proper SemVer practices, users might face unexpected issues or be unable to safely upgrade their dependencies, which can result in losing the user.
+
+## Common Mistakes to Avoid
+
+### Incorrect Versioning
+Ensure you understand the type of changes you’re making. For example, if you introduce breaking changes but incorrectly release it as a patch update (e.g., from `1.0.0` to `1.0.1`), it can cause significant issues for users relying on version ranges like `^1.0.0` or `~1.0.0` in their `package.json`.  
+
+These ranges automatically pull in updates for compatible versions. By incorrectly marking a breaking change as a patch, you risk breaking their projects without warning, damaging trust in your package. Always increment the **MAJOR** version for breaking changes to ensure users can consciously decide when to adopt them.  
+
+To better understand what the `^`, `~`, and other symbols mean in npm versioning, check out [What do the different symbols mean for npm version?](https://www.ssw.com.au/rules/different-symbols-mean-for-npm-version).  
+
+### Forgetting to Communicate Breaking Changes
+For major updates, clearly communicate the breaking changes in your release notes or changelog. This helps users prepare for or adapt to the changes.
+
+## Tools to Help You Follow SemVer
+
+Here are a few tools and resources that can help you manage SemVer compliance when publishing npm packages:
+
```

Comment:
I think add changeset lib as the first point here (since we use it on TInacms) and its made by Aussies :P